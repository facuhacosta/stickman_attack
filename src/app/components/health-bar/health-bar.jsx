import React from 'react';
import style from './health-bar.module.scss';
  
export function HealthBar({ completed, health }) {

  return (
    <div className={style['health-bar']}>
      <div className={style['fillerStyles']} style={{width: `${completed}%`}}>
      </div>
      <span className={style['labelStyles']}>{`${health}`}</span>
    </div>
  );
}