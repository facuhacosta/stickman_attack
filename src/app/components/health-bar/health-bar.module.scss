.health-bar {
  height: 10px;
  width: 100%;
  background-color: #e0e0de;
  border-radius: 5px;
  text-align: center;
  font-size: x-small;

  .fillerStyles {
    height: 100%;
    background-color: green;
    border-radius: inherit;

  }
  .labelStyles {
    padding: 5;
    color: white;
    font-weight: bold;
    position: relative;
    top: -110%;
  }
}