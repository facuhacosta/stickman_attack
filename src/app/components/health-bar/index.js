export { HealthBar } from './health-bar';