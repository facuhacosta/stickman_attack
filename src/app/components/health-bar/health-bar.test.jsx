import React from 'react';
import { render } from '@testing-library/react';
import { HealthBar } from './health-bar';

describe('HealthBar Component', () => {

  // Matching Snapshots
  test('it should match the snapshot', () => {
    const { asFragment } = render(<HealthBar />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Unit tests

});