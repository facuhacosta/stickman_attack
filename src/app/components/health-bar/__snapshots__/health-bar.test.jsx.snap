// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthBar Component it should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div />
    <span>
      undefined
    </span>
  </div>
</DocumentFragment>
`;
